scripts/knot_benchmark/src/benchmark/cases.py

```diff
@@ -0,0 +1,212 @@
+            "--cache-dir",
+            os.devnull,
+        ]
+
+        return Command(
+            name="mypy",
+            command=command,
+        )
+
+    def warm_command(self, project: Project, venv: Venv) -> Command | None:
+        command = [
+            str(self.path),
+            *(project.mypy_arguments or project.include),
+            "--python-executable",
+            str(venv.python),
+        ]
+
+        return Command(
+            name="mypy",
+            command=command,
+        )
+
+    def _base_command(self, project: Project, venv: Venv) -> list[str]:
+        return [
+            str(self.path),
+            "--python-executable",
+            str(venv.python),
+            *(project.mypy_arguments or project.include),
+        ]
+
+
+class Pyright(Tool):
+    path: Path
+
+    def __init__(self, *, path: Path | None = None):
+        self.path = path or which_tool("pyright")
+
+    def cold_command(self, project: Project, venv: Venv) -> Command:
+        command = [
+            str(self.path),
+            "--venvpath",
+            str(
+                venv.path.parent
+            ),  # This is not the path to the venv folder, but the folder that contains the venv...
+            *(project.pyright_arguments or project.include),
+        ]
+
+        return Command(
+            name="Pyright",
+            command=command,
+        )
+
+
+class Venv:
+    path: Path
+
+    def __init__(self, path: Path):
+        self.path = path
+
+    @property
+    def name(self):
+        """The name of the virtual environment directory."""
+        return self.path.name
+
+    @property
+    def python(self):
+        """Returns the path to the python executable"""
+        return self.script("python")
+
+    @property
+    def bin(self) -> Path:
+        bin_dir = "scripts" if sys.platform == "win32" else "bin"
+        return self.path / bin_dir
+
+    def script(self, name: str) -> Path:
+        extension = ".exe" if sys.platform == "win32" else ""
+        return self.bin / f"{name}{extension}"
+
+    @staticmethod
+    def create(parent: Path) -> Venv:
+        """Creates a new, empty virtual environment."""
+
+        command = [
+            "uv",
+            "venv",
+            "--quiet",
+            "venv",
+        ]
+
+        try:
+            subprocess.run(
+                command, cwd=parent, check=True, capture_output=True, text=True
+            )
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(f"Failed to create venv: {e.stderr}")
+
+        root = parent / "venv"
+        return Venv(root)
+
+    def install(self, dependencies: list[str]):
+        """Installs the dependencies required to type check the project."""
+
+        logging.debug(f"Installing dependencies: {', '.join(dependencies)}")
+        command = [
+            "uv",
+            "pip",
+            "install",
+            "--quiet",
+            *dependencies,
```

Comment:
Worth adding `--exclude-newer`? While you pin the commit, you don't pin the deps